x-pack/auditbeat/processors/sessionmd/add_session_metadata.go

```diff
@@ -0,0 +1,215 @@
+	_, err := ev.GetValue(p.config.PIDField)
+	if err != nil {
+		// Do not attempt to enrich events without PID; it's not a supported event
+		return ev, nil //nolint:nilerr // Running on events without PID is expected
+	}
+
+	err = p.provider.UpdateDB(ev)
+	if err != nil {
+		return ev, err
+	}
+
+	result, err := p.enrich(ev)
+	if err != nil {
+		return ev, fmt.Errorf("enriching event: %w", err)
+	}
+	return result, nil
+}
+
+func (p *addSessionMetadata) String() string {
+	return fmt.Sprintf("%v=[backend=%s, pid_field=%s, replace_fields=%t]",
+		processorName, p.config.Backend, p.config.PIDField, p.config.ReplaceFields)
+}
+
+func (p *addSessionMetadata) enrich(ev *beat.Event) (*beat.Event, error) {
+	pidIf, err := ev.GetValue(p.config.PIDField)
+	if err != nil {
+		return nil, err
+	}
+	pid, err := pidToUInt32(pidIf)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse pid field '%s': %w", p.config.PIDField, err)
+	}
+
+	fullProcess, err := p.db.GetProcess(pid)
+	if err != nil {
+		return nil, fmt.Errorf("pid %v not found in db: %w", pid, err)
+	}
+
+	result := ev.Clone()
+
+	processMap := fullProcess.ToMap()
+
+	if b, err := result.Fields.HasKey("process"); !b || err != nil {
+		return nil, fmt.Errorf("no process field in event")
+	}
+	m, ok := tryToMapStr(result.Fields["process"])
+	if !ok {
+		return nil, fmt.Errorf("process field type not supported")
+	}
+
+	err = mapstr.MergeFieldsDeep(m, processMap, true)
+	if err != nil {
+		return nil, fmt.Errorf("merging enriched fields with event: %w", err)
+	}
+
+	if p.config.ReplaceFields {
+		if err := p.replaceFields(result); err != nil {
+			return nil, fmt.Errorf("replace fields: %w", err)
+		}
+	}
+	return result, nil
+}
+
+// pidToUInt32 converts PID value to uint32
+func pidToUInt32(value interface{}) (pid uint32, err error) {
+	switch v := value.(type) {
+	case string:
+		nr, err := strconv.Atoi(v)
+		if err != nil {
+			return 0, fmt.Errorf("error converting string to integer: %w", err)
+		}
+		pid = uint32(nr)
+	case uint32:
+		pid = v
+	case int, int8, int16, int32, int64:
+		pid64 := reflect.ValueOf(v).Int()
+		if pid = uint32(pid64); int64(pid) != pid64 {
+			return 0, fmt.Errorf("integer out of range: %d", pid64)
+		}
+	case uint, uintptr, uint8, uint16, uint64:
+		pidu64 := reflect.ValueOf(v).Uint()
+		if pid = uint32(pidu64); uint64(pid) != pidu64 {
+			return 0, fmt.Errorf("integer out of range: %d", pidu64)
+		}
+	default:
+		return 0, fmt.Errorf("not an integer or string, but %T", v)
+	}
+	return pid, nil
+}
+
+// replaceFields replaces event fields with values suitable user with the session viewer in Kibana
+// The current version of session view in Kibana expects different values than what are used by auditbeat
+// for some fields. This function converts these field to have values that will work with session view.
+//
+// This function is temporary, and can be removed when Kibana is updated to work with the auditbeat field values.
```

Comment:
"temporary" can quickly lose its meaning once checked in. Can you please reference a GH issue so that it's obvious when this temporary function can be removed.

Reply:
I've added the issue number for the Kibana work on session viewer here